Task07/script.js

```diff
@@ -0,0 +1,47 @@
+(() => {
+'use strict';
+	
+	//ドロップダウン内の製品と対応する製品コードの定義
+	const dropdownData = {
+		'kintone': 'KN',
+		'Garoon': 'GR',
+		'サイボウズ Office': 'OF',
+		'Mailwise': 'MW'
+	};
+
+	const avoidDuplication = (event) => {
+		//日付からハイフンを抜いた文字列を作成
+		const dateData =() => {
+			const dateStr = event.record.日付.value;	
+			return dateStr.replace(/-/g, '');
+		} ;
+
+		//ドロップダウンで選択された製品に対応する製品コードの文字列を作成
+		const productCode = dropdownData[event.record.サイボウズ製品.value];
+
+		//入力された管理番号の文字列を作成
+		const manageNumber = event.record.管理番号.value;
+
+		//重複禁止項目に入力する文字列の作成
+		event.record.重複禁止項目_文字列.value = dateData() + '-' + productCode + '-' + manageNumber;
```

Comment:
提案（修正しなくてもOK）
テンプレートリテラルを使うと `+` を書かなくて済みます！
```suggestion
		event.record.重複禁止項目_文字列.value = `${dateData()}-${productCode}-${manageNumber}`
```

Reply:
クエリでバッククォートの有能さを知りました、、
+を書かないだけで楽になりましたが、今回のケースだと変数間の文字がハイフンなので少し見にくいのかなとも思いました！時と場合で使い分けていきます。